### PR TITLE
Support discontinuous fluxes in elliptic DG

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Three changes to the elliptic DG scheme, mainly to support elasticity with layered materials as published in https://arxiv.org/abs/2111.06893. See commit messages for details.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: tests/Unit/Elliptic/DiscontinuousGalerkin/Test_DgOperator.cpp [hunk 1]
```diff
@@ -1107,6 +1109,141 @@ SPECTRE_TEST_CASE("Unit.Elliptic.DG.Operator", "[Unit][Elliptic]") {
...
+              elliptic::BoundaryConditions::AnalyticSolution<system>>(
+              analytic_solution.get_clone(),
+              elliptic::BoundaryConditionType::Dirichlet)};
+      const ElementId<3> center_id{6};
+      const ElementId<3> wedge_id{0};
+      using Vars = Variables<tmpl::list<Var<Poisson::Tags::Field>>>;
+      using OperatorVars =
+          Variables<tmpl::list<DgOperatorAppliedTo<Var<Poisson::Tags::Field>>>>;
+      Vars vars_rnd_center{27};
+      get(get<Var<Poisson::Tags::Field>>(vars_rnd_center)) = DataVector{
```
Comment (nilsdeppe) [inline tests/Unit/Elliptic/DiscontinuousGalerkin/Test_DgOperator.cpp:1142]: Can you make these approximate comparisons? We will almost inevitably run on different hardware or compilers that will break exact comparison.

**Comment by nilsvu** [comment]
@nilsdeppe would you be able to review this?

**Comment by nilsdeppe** [comment]
Sure, once it passes CI I can look

**Comment by nilsvu** [comment]
CI should be fixed now. Either way, can you do #5791 first? I rebased on that to have the extra regression tests here.

File: tests/InputFiles/Elasticity/SingleCoatingMirror.yaml [hunk 2]
```diff
@@ -0,0 +1,174 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+Executable: SolveElasticity3D
```
Comment (nilsdeppe) [inline tests/InputFiles/Elasticity/SingleCoatingMirror.yaml:4]: Does this mean we build yet another executable in CI? If so, can we replace an existing build instead?

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
I don't have a concrete suggestion here, just a thing to note comparing to the evolution code. We remove a lot of memory allocations that used to be done on interfaces/mortars, which helped quite a bit with performance in multithreaded cases

File: tests/Unit/Elliptic/DiscontinuousGalerkin/Test_DgOperator.cpp (hunk 1 above)
Comment (nilsvu) [reply tests/Unit/Elliptic/DiscontinuousGalerkin/Test_DgOperator.cpp:1142]: The comparisons are done with our standard approx. Here I just assign the random test data, the comparisons are elsewhere in this file.

File: tests/InputFiles/Elasticity/SingleCoatingMirror.yaml (hunk 2 above)
Comment (nilsvu) [reply tests/InputFiles/Elasticity/SingleCoatingMirror.yaml:4]: No the HalfSpaceMirror.yaml already builds this exec.

**Comment by nilsvu** [comment]
Re memory allocations: there's a comment at the start of the function about that. I'll deal with it when I do a performance optimization pass over the elliptic code.

**Comment by nilsvu** [comment]
Rebased because two CI tests failed in an unrelated way

**Review by nilsdeppe — APPROVED** [review]
Sounds good!

**Comment by nilsvu** [comment]
CI failure looks like #5832

**Pull request merged**